internal/cli/federatedauthentication/federationsettings/identityprovider/revokejwk.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package identityprovider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/templatewriter"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115012/admin"
+)
+
+type RevokeOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	store                store.IdentityProviderJwkRevoker
+	FederationSettingsID string
+	IdentityProviderID   string
+}
+
+const revokeTemplate = `Identity Provider {{.Id}} JWKs tokens revoked.`
+
+func (opts *RevokeOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *RevokeOpts) Run() error {
+	err := opts.store.RevokeJwksFromIdentityProvider(opts.FederationSettingsID, opts.IdentityProviderID)
+	if err != nil {
+		return err
+	}
+
+	return templatewriter.Print(opts.ConfigWriter(), revokeTemplate, atlasv2.FederationIdentityProvider{Id: opts.IdentityProviderID})
```

Comment:
The reason why the template check fails is because of this line.

All other commands which use templates use `opts.Print` instead of calling `templatewriter` directly.
Is it possible to modify your code to this instead?

```go
federationProvider := atlasv2.FederationIdentityProvider{Id: opts.IdentityProviderID}
return opts.Print(federationProvider)
```

Or is there a reason why you call the `templatewriter` directly?
If so I'll update the template check for you to support this scenario too.

Reply:
we synced offline and removed this completely, thanks!